### PR TITLE
root: add determine_variants

### PR DESCRIPTION
**Pull request by vvolkl**
mostly serves to increase the size of this  too-short recipe `</joke>`. 

The variants were obviously not written with achieving a one-to-one mapping to roots build options and that's probably a good thing, because mainly they should be for installing root with spack. Even if it is not 100% for this reason, being a bit more accurate when running `spack external find root`  may still prove useful in some scenarios.

### CODE REVIEW

**Comment by HadrienG2** [comment]
How strange, it seems I am not able to review PRs in this repository anymore. Maybe some issue with the recent permissions changes?

Let's ping @adamjstewart for help.

In any case, I am happy with those changes.

File: var/spack/repos/builtin/packages/root/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/root/package.py:332]: "Which" is duplicated in the comment

File: var/spack/repos/builtin/packages/root/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/root/package.py]: You should use exes here, as there may be multiple in the PATH

File: var/spack/repos/builtin/packages/root/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/root/package.py]: Split without args defaults to whitespace

File: var/spack/repos/builtin/packages/root/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/root/package.py]: There's a lot of duplication here. I would use a dict or list and loop over the entries instead

File: var/spack/repos/builtin/packages/root/package.py
Comment (vvolkl) [reply var/spack/repos/builtin/packages/root/package.py]: Ah, I was a bit confused by that initially since the interface is different from determine_version, but I think I understand it now.

File: var/spack/repos/builtin/packages/root/package.py
Comment (vvolkl) [reply var/spack/repos/builtin/packages/root/package.py]: Hmmm, I'm not sure that makes a lot of difference. In the end, I need to write out the variant and feature names, and if I put that in a list or one macro-like function call per line is basically the same. In the future maybe we could refactor things to have one list of variants  and build flags in the recipe  that can be used in different places.

I did think about adding a `nonlocal` to `add_variant_from_feature`, to eliminate the first two arguments, but that doesn't work with python 2 I think. So as a compromise I just choose some more concise names for the repeated things.

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Waiting for other maintainers to review.

**Review by HadrienG2 — APPROVED** [review]
Looks good to me. Thanks for the permission fix @adamjstewart!

**Review by alalazo — APPROVED** [review]

**Pull request merged**
